src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/PackageRestore/PackageRestoreDataSource.cs

```diff
@@ -206,5 +222,67 @@ public Task UnloadAsync()
 
             return Task.CompletedTask;
         }
+
+        // NuGet calls this method to wait project to nominate restoring
+        // If the project has no pending restore data, it will return a completed task.
+        // Otherwise a task which will be completed once the project norminate the next restore
+        // the task will be cancelled, if the project system decide it no longer need restore (for example: the restore state has no change)
+        // the task will be failed, if the project system runs into a problem, so it cannot get correct data to norminate a restore (DT build failed)
+        public Task WhenNominated(CancellationToken cancellationToken)
+        {
+            if (_hasPendingNomination == false || cancellationToken.IsCancellationRequested)
+            {
+                return Task.CompletedTask;
+            }
+
+            JoinableTask joinableTask = JoinableFactory.RunAsync(() =>
+            {
+                while (true)
```

Comment:
This is burning CPU waiting for stuff, please work with @lifengl on an approach that doesn't do this.

Reply:
fixed